src/main/java/io/cdap/plugin/http/common/pagination/BaseHttpPaginationIterator.java

```diff
@@ -83,11 +83,13 @@ public BaseHttpPaginationIterator(BaseHttpSourceConfig config, PaginationIterato
   protected boolean visitPageAndCheckStatusCode() throws IOException {
     if (response != null) { // close previous response
       response.close();
+      getHttpClient().close();
```

Comment:
Creation of new http client for every request seems to be inefficient, is it possible to just set new auth token in header with existing http client?

Reply:
HttpClient does not provide, any API to update the AuthenticationHeader. But let me check if there is any other way to reuse the HttpClient object created here.